diff --git a/dist/index.d.ts b/dist/index.d.ts
index db446d06530025a016587080bf2f773e19ba7386..639b00d0ae2e2b56fe7d48396ba0a2b2e8405627 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -1,4 +1,4 @@
-import { z } from 'astro/zod';
+import { z } from 'zod';
 import { rssSchema } from './schema.js';
 export { rssSchema };
 export type RSSOptions = {
diff --git a/dist/index.js b/dist/index.js
index d5d9eab7280597a43853cc62588c7d079805f858..27832522cdba442cb03f99acc9e494e015785848 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -1,4 +1,4 @@
-import { z } from "astro/zod";
+import { z } from "zod";
 import { XMLBuilder, XMLParser } from "fast-xml-parser";
 import { yellow } from "kleur/colors";
 import { rssSchema } from "./schema.js";
diff --git a/dist/schema.d.ts b/dist/schema.d.ts
index 1e82106cff9268c0669051dd8908a54fe6ffb7c0..62a1b208be914efc48798f484e6ab833dcaec5b3 100644
--- a/dist/schema.d.ts
+++ b/dist/schema.d.ts
@@ -1,4 +1,4 @@
-import { z } from 'astro/zod';
+import { z } from 'zod';
 export declare const rssSchema: z.ZodObject<{
     title: z.ZodOptional<z.ZodString>;
     description: z.ZodOptional<z.ZodString>;
diff --git a/dist/schema.js b/dist/schema.js
index 026882b1bbff44049ea1e5a563d3c68c1ba5e353..61251e024b8bdc33dcd105a2739d9ea5c8a0f337 100644
--- a/dist/schema.js
+++ b/dist/schema.js
@@ -1,4 +1,4 @@
-import { z } from "astro/zod";
+import { z } from "zod";
 const rssSchema = z.object({
   title: z.string().optional(),
   description: z.string().optional(),
diff --git a/dist/util.d.ts b/dist/util.d.ts
index dbfc751edcc989120aada993821f93ec6ee073b4..33da014f95e870c937ac4f9636dcd268bd5ae401 100644
--- a/dist/util.d.ts
+++ b/dist/util.d.ts
@@ -1,4 +1,4 @@
-import type { z } from 'astro/zod';
+import type { z } from 'zod';
 import type { RSSOptions } from './index.js';
 /** Normalize URL to its canonical form */
 export declare function createCanonicalURL(url: string, trailingSlash?: RSSOptions['trailingSlash'], base?: string): string;
