diff --git a/bin/workerd b/bin/workerd
index 9ef8f56faaa6015a1023af88f06720557bce72dc..294a3aadc73b82effee5a49b571e0ecfe4572ac9 100644
--- a/bin/workerd
+++ b/bin/workerd
@@ -31,7 +31,8 @@ var knownPackages = {
   "darwin x64 LE": "@cloudflare/workerd-darwin-64",
   "linux arm64 LE": "@cloudflare/workerd-linux-arm64",
   "linux x64 LE": "@cloudflare/workerd-linux-64",
-  "win32 x64 LE": "@cloudflare/workerd-windows-64"
+  "win32 x64 LE": "@cloudflare/workerd-windows-64",
+  "win32 arm64 LE": "@cloudflare/workerd-windows-64"
 };
 var maybeExeExtension = process.platform === "win32" ? ".exe" : "";
 function pkgAndSubpathForCurrentPlatform() {
diff --git a/install.js b/install.js
index bd6607fed7fd745cf49fb9423965be0204cb8fe6..ec529e814c5a6d8ca8aa313f2e6838e0aca43af3 100644
--- a/install.js
+++ b/install.js
@@ -29,7 +29,8 @@ var knownPackages = {
   "darwin x64 LE": "@cloudflare/workerd-darwin-64",
   "linux arm64 LE": "@cloudflare/workerd-linux-arm64",
   "linux x64 LE": "@cloudflare/workerd-linux-64",
-  "win32 x64 LE": "@cloudflare/workerd-windows-64"
+  "win32 x64 LE": "@cloudflare/workerd-windows-64",
+  "win32 arm64 LE": "@cloudflare/workerd-windows-64"
 };
 var maybeExeExtension = process.platform === "win32" ? ".exe" : "";
 function pkgAndSubpathForCurrentPlatform() {
diff --git a/lib/main.js b/lib/main.js
index 584241bb15db752a9b88867b14e639925a659a20..60799e9351e71b6684231e3e56e49ed405ef36c4 100644
--- a/lib/main.js
+++ b/lib/main.js
@@ -45,7 +45,8 @@ var knownPackages = {
   "darwin x64 LE": "@cloudflare/workerd-darwin-64",
   "linux arm64 LE": "@cloudflare/workerd-linux-arm64",
   "linux x64 LE": "@cloudflare/workerd-linux-64",
-  "win32 x64 LE": "@cloudflare/workerd-windows-64"
+  "win32 x64 LE": "@cloudflare/workerd-windows-64",
+  "win32 arm64 LE": "@cloudflare/workerd-windows-64"
 };
 var maybeExeExtension = process.platform === "win32" ? ".exe" : "";
 function pkgAndSubpathForCurrentPlatform() {
