diff --git a/dist/index.mjs b/dist/index.mjs
index 8f3fe7d619f247640705232ab9491c12f3e84979..a5b75f0fe574672f5714cfa866762106d044e9b0 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -2,6 +2,7 @@ import { ShikiError as ShikiError$1 } from "@shikijs/types";
 import { ShikiError, applyColorReplacements, codeToTokensBase as codeToTokensBase$1, codeToTokensWithThemes, codeToTokensWithThemes as codeToTokensWithThemes$1, createShikiInternal, createShikiInternalSync, createShikiPrimitive, createShikiPrimitive as createShikiPrimitive$1, createShikiPrimitiveAsync, createShikiPrimitiveAsync as createShikiPrimitiveAsync$1, getLastGrammarState, getLastGrammarStateFromMap, isNoneTheme, isPlainLang, isSpecialLang, isSpecialTheme, normalizeGetter, normalizeTheme, resolveColorReplacements, setLastGrammarStateToMap, splitLines, splitLines as splitLines$1, toArray, tokenizeWithTheme } from "@shikijs/primitive";
 import { FontStyle } from "@shikijs/vscode-textmate";
 import { toHtml } from "hast-util-to-html";
+import { highlightTokens, fromAlias } from "@master-hash/tree-sitter-highlight";
 export * from "@shikijs/types";
 //#region src/utils/hast.ts
 const RE_WHITESPACE = /\s+/g;
@@ -693,6 +694,9 @@ function dimColor(color) {
 * This wraps the tokenizer's implementation to add ANSI support.
 */
 function codeToTokensBase(primitive, code, options = {}) {
+	const _a = fromAlias(options.lang);
+	// if (_a !== null) { console.log(highlightTokens(code, _a).tokens[0]); }
+	if (_a !== null) { return highlightTokens(code, _a).tokens; }
 	const lang = primitive.resolveLangAlias(options.lang || "text");
 	const { theme: themeName = primitive.getLoadedThemes()[0] } = options;
 	if (!isPlainLang(lang) && !isNoneTheme(themeName) && lang === "ansi") {
@@ -715,7 +719,14 @@ function codeToTokens(primitive, code, options) {
 	let themeName;
 	let rootStyle;
 	let grammarState;
-	if ("themes" in options) {
+	if (fromAlias(options.lang) !== null) {
+		tokens = codeToTokensBase(
+			primitive,
+			code,
+			options
+		);
+		// console.log(tokens);
+	} else if ("themes" in options) {
 		const { defaultColor = "light", cssVariablePrefix = "--shiki-", colorsRendering = "css-vars" } = options;
 		const themes = Object.entries(options.themes).filter((i) => i[1]).map((i) => ({
 			color: i[0],
